gtsfm/frontend/matcher/image_matcher_base.py

```diff
@@ -0,0 +1,55 @@
+"""Base class for front end matchers that directly identify image correspondences, without using explicit descriptors.
+
+Authors: John Lambert
+"""
+import abc
+from typing import Tuple
+
+import dask
+from dask.delayed import Delayed
+
+from gtsfm.common.image import Image
+from gtsfm.common.keypoints import Keypoints
+
+
+class ImageMatcherBase(metaclass=abc.ABCMeta):
+    """Base class for matchers that accept an image pair, and immediately generate keypoint matches.
+
+    Note: these matchers do NOT use descriptors as input.
+    """
+
+    @abc.abstractmethod
+    def match(
+        self,
+        image_i1: Image,
+        image_i2: Image,
+    ) -> Tuple[Keypoints, Keypoints]:
+        """Identify feature matches across two images.
+
+        If the results are in the cache, they are fetched and returned. Otherwise, the `match()` of the
```

Comment:
Docstring seems to be incorrectly copied from the cacher

Reply:
Thanks for the review Ayush. Could you explain a bit more about which part of this docstring seemed out of place?

Reply:
>  If the results are in the cache, they are fetched and returned. Otherwise,....

Reply:
Hmm this description is correct, right?

Reply:
The base class need not know anything about the cache, right? Its a simple API which takes an image and "produces" the matches. Whether it is using a cache is the detail of the cacher, and not of the base API

Reply:
Oh you're absolutely right, thanks. Updated now.